agent/structs/connect_proxy_config.go

```diff
@@ -157,6 +157,23 @@ func (u *Upstream) ToAPI() api.Upstream {
 	}
 }
 
+// Identifier returns a string representation that identifies the upstream in an
+// unambiguous but human readable way.
+func (u *Upstream) Identifier() string {
```

Comment:
Do we want to call this `String() string` and just implement `stringer` or no?

Reply:
I thought about that. I don't feel especially strongly but we rely on this being unique in a bunch of places (used as a map key for example) so the name `Identifier` seemed to make that more explicit than just "a string representation". I don't feel strongly though - the doc comment can state that it guarantees to be uniquely identifyingI suppose.

Reply:
I think its good as is, if anything you may want to just add the `String() string { return u.Identifier() }` implementation so that `%s` fmting works with this struct.